kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/TypeRenderer.kt

```diff
@@ -43,6 +43,16 @@ abstract class TypeRenderer {
 
   fun render(typeName: TypeName): CodeBlock {
     if (typeName.nullable) {
+      if (typeName == BOOLEAN.asNullable()
+          || typeName == BYTE.asNullable()
+          || typeName == CHAR.asNullable()
+          || typeName == DOUBLE.asNullable()
+          || typeName == FLOAT.asNullable()
+          || typeName == INT.asNullable()
+          || typeName == LONG.asNullable()
+          || typeName == SHORT.asNullable()) {
```

Comment:
`typeName.isBoxedPrimitive()` should work here

Reply:
that exists in JavaPoet. is there something for KotlinPoet? i'm a little behind.

Reply:
Oh right. I've spent the week with JavaPoet and it's on the mind.